python/pyspark/sql/dataframe.py

```diff
@@ -422,6 +422,63 @@ def repartition(self, numPartitions):
         """
         return DataFrame(self._jdf.repartition(numPartitions), self.sql_ctx)
 
+    @since(1.6)
```

Comment:
This should be `1.3`, we can have `.. versionchanged:: version xxx` in the end of doc.

In order to have good order, we may not use `since` here, but
```
.. versionadd:: 1.3
.. versionchanged:: 1.6 xxx
```